### ISSUE TITLE
Clone folder cannot be created outside of $HOME directory

### ISSUE THREAD

**Issue by FeodorFitsner** [appveyor/ci#2142]
The following will fail to create a custom clone folder:

```
clone_folder: /my_clone_folder
```

Correct script:

```bash
sudo mkdir $clone_folder
sudo chown -R $(id -u):$(id -g) $clone_folder
```

Correct cleanup code:

```bash
sudo rm -rf $clone_folder
```

**Comment by SergeGautherie** [comment]
`/my_clone_folder` is fixed.

But `/my_first_folder/my_clone_folder` still has #2305

**Comment by Wasapl** [comment]
Our tests can't reproduce same issue.
Can you please share your appveyor.yml with us? You may send it via email team@appveyor.com in case you you dont want public it worldwide.

**Comment by SergeGautherie** [comment]
Result: [Test build 408](https://ci.appveyor.com/project/SergeGautherie/reactos/build/reactos.appveyor.408)
NB: Ignore `git checkout` unrelated issue.

--

Test `appveyor.yml`:
```yaml
image: Ubuntu

version: reactos.appveyor.{build}
skip_branch_with_pr: true
clone_depth: 5
clone_folder: /my_first_folder/my_clone_folder

init:
  - sh: uname -a
  - sh: pwd
  - sh: cd /my_first_folder
  - sh: ls -la

test: off
deploy: off
```

**Comment by Wasapl** [comment]
Oh I see. So, in order to create clone folder in Root directory `/` one have to have root privileges. Therefore we create clone folder with sudo like `sudo mkdir -p '/my_first_folder/my_clone_folder' and change owner to appveyor for most inner one  `my_clone_folder`. Then we `cd` in it and  clone repository in it.
We assume that your script do all that you need inside of `my_clone_folder`. If you need to do something in parent folder `my_first_folder` you always can do something like `sudo chown appveyor:appveyor /my_first_folder` among all other stuff you need.
On the other hand creating folder inside `$HOME` (which in our case will be `/home/appveyor`) directory does not require root privileges. If you provide `clone_folder: my_first_folder/my_clone_folder` in appveyor.yml we assume that you want us to create clone folder inside `$HOME` directory i.e `/home/appveyor/my_first_folder/my_clone_folder` and we create it without sudo. 
Summing up: 
1. this is as intended.
2. you may create clone folder in $HOME directory like `clone_folder: my_first_folder/my_clone_folder`. Just delete leading slash.
3. you may change owner of my_first_folder in init stage e.g ` sudo chown appveyor:appveyor /my_first_folder`

We eager to hear your thoughts about it!

**Comment by SergeGautherie** [comment]
1. I understand this behavior is intended as administrator. But it feels unexpected as user.
2. That was issue 2305, but is not the current issue.
3. Yes, I can "work around" it myself, though I would expect that to be done automatically:
as in created directories belong to `appveyor` user, while existing directories remain as they are.
